consensus/src/quorum_store/tests/utils.rs

```diff
@@ -317,94 +319,95 @@ fn test_proof_calculate_remaining_txns_and_proofs() {
     assert_eq!(proof_queue.batch_summaries_len(), 1);
 
     // Adding an expired batch again
-    proof_queue.add_batch_summaries(vec![(info_3, vec![txns[0]])]);
+    proof_queue.insert_batches(vec![(info_3, vec![txns[0]])]);
     // txns: [txn_1] + txns(proof_6)
     // proofs: [2, 6]
     // batch_summaries: [2 -> txn_1, 3 -> txn_0]
     assert_eq!(proof_queue.remaining_txns_and_proofs(), (2, 2));
     assert_eq!(proof_queue.batch_summaries_len(), 2);
 
     // Adding an expired proof again. Should have no effect
-    proof_queue.push(author_0_batches[2].clone());
+    proof_queue.insert_proof(author_0_batches[2].clone());
     // txns: [txn_1] + txns(proof_6)
     // proofs: [2, 6]
     // batch_summaries: [2 -> txn_1, 3 -> txn_0]
     assert_eq!(proof_queue.remaining_txns_and_proofs(), (2, 2));
     assert_eq!(proof_queue.batch_summaries_len(), 2);
 
-    proof_queue.add_batch_summaries(vec![(info_7, vec![txns[3]])]);
+    proof_queue.insert_batches(vec![(info_7, vec![txns[3]])]);
     // txns: [txn_1] + txns(proof_6)
     // proofs: [2, 6]
     // batch_summaries: [2 -> txn_1, 7 -> txn_3, 3 -> txn_0]
     assert_eq!(proof_queue.remaining_txns_and_proofs(), (2, 2));
     assert_eq!(proof_queue.batch_summaries_len(), 3);
 
     proof_queue.handle_updated_block_timestamp(now_in_usecs + 30000);
-    // Expires info_6
+    // Expires info_6, info_3
     // txns: [txn_1]
     // proofs: [2]
-    // batch_summaries: [2 -> txn_1, 7 -> txn_3, 3 -> txn_0]
+    // batch_summaries: [2 -> txn_1, 7 -> txn_3]
```

Comment:
Why is info_3 expiring now but not before?

Reply:
`info_3` corresponds to this `proof_of_store(author_0, BatchId::new_for_test(2), 50, now_in_usecs + 20000)`.

Since we are updating block timestamp to `now() +30000` then info_3 should be expired and removed right

Reply:
There is no change in the batches or transactions that are inserted in this unit test. But the result has changed. Either this PR fixed an existing bug, or this PR introduced a new bug. Wondering which is the case.

Reply:
There is a slight logic change, but for better. Batch summaries were never cleaned up before based on block timestamp which was not asserted in unit testing previously, but now it is.